src/transactions/LiquidityPoolDepositOpFrame.cpp

```diff
@@ -120,8 +118,7 @@ LiquidityPoolDepositOpFrame::depositIntoNonEmptyPool(
                               cp.totalPoolShares, ROUND_UP);
         if (!(resA && resB))
         {
-            innerResult().code(LIQUIDITY_POOL_DEPOSIT_UNDERFUNDED);
-            return false;
+            throw std::runtime_error("amount overflowed");
```

Comment:
Can you add a comment here explaining why this can't happen? Something along the lines of:

Neither amount can overflow because
```
amountA  = ceil(amountPoolShares * reserveA / totalPoolShares)
         = ceil(min(sharesA, sharesB) * reserveA / totalPoolShares)
        <= ceil(sharesA * reserveA / totalPoolShares)
         = ceil(floor(totalPoolShares * maxAmountA / reserveA) * reserveA / totalPoolShares)
        <= ceil(totalPoolShares * maxAmountA / reserveA * reserveA / totalPoolShares) 
         = maxAmountA .
```
and everything works analogously for `amountB`.

Reply:
Done

Reply:
Nitpick: Everything works analogously appears twice

Reply:
Fixed